### PR TITLE
apis: add elastic quota profile

### PR DESCRIPTION
**Pull request by shaloulcy**
### Ⅰ. Describe what this PR does

add elastic quota profile

Every quota profile will generate a root quota tree

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: apis/quota/v1alpha1/elastic_quota_profile_types.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+type ElasticQuotaProfileSpec struct {
+	// RootQuotaName defines the root quota name of the profile.
+	// +required
+	RootQuotaName string `json:"rootQuotaName"`
+	// RootQuotaLabels defines the labels of the root quota.
+	RootQuotaLabels map[string]string `json:"rootQuotaLabels,omitempty"`
+	// NodeSelector defines a node selector to select nodes.
+	// +required
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
```
Comment (FillZpp) [inline apis/quota/v1alpha1/elastic_quota_profile_types.go:34]: require field should not be `omitempty`

File: apis/quota/v1alpha1/elastic_quota_profile_types.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2022 The Koordinator Authors.
```
Comment (FillZpp) [inline apis/quota/v1alpha1/elastic_quota_profile_types.go:2]: All new files should be `2023`

Comment (shaloulcy) [reply apis/quota/v1alpha1/elastic_quota_profile_types.go:2]: the license-check will check it and reject it

**Review by eahydra — COMMENTED** [review]
/lgtm

File: apis/quota/v1alpha1/elastic_quota_profile_types.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type ElasticQuotaProfileSpec struct {
+	// RootQuotaName defines the root quota name of the profile.
+	// +required
+	RootQuotaName string `json:"rootQuotaName"`
```
Comment (hormes) [inline apis/quota/v1alpha1/elastic_quota_profile_types.go:29]: Remove the root, this is the meaning given to it by the outside.

**Review by jasonliu747 — APPROVED** [review]

**Comment by hormes** [comment]
#1606

**Pull request merged**